### PR TITLE
performance: do not synchronize JavaModelManager.temporaryCache #1894

### PR DESCRIPTION
**Pull request by jukzi**
used in JavaModelManager.getInfo(IJavaElement)
it's a ThreadLocal anyway.

https://github.com/eclipse-jdt/eclipse.jdt.core/issues/1894

### CODE REVIEW

File: org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaModelManager.java [hunk 1]
```diff
@@ -1258,7 +1257,7 @@ public static IJavaElement determineIfOnClasspath(IResource resource, IJavaProje
 	/**
 	 * Infos cache.
 	 */
-	private JavaModelCache cache;
+	private final JavaModelCache cache = new JavaModelCache();
```
Comment (iloveeclipse) [inline org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaModelManager.java:1260]: Interesting, this is actually "back to the roots". Original code had this until commit af33da7216734bcb2ae0e5dbadaa836c42812db8 that moved init after init of debug options.

**Review by iloveeclipse — APPROVED** [review]
Looks safe. 

Do you see a measurable performance improvement with this?

File: org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaModelManager.java (hunk 1 above)
Comment (jukzi) [reply org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaModelManager.java:1260]: The JavaModelCache constructor already uses the debug options, therefore the debug options have been initialized before.

**Comment by jukzi** [comment]
closing because i could not reproduce + issue with debug tracing options

**Pull request closed without merging**